sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -0,0 +1,175 @@
+    /**
+     * Given a context containing the current tracing span the span is marked completed with status info from
+     * {@link Signal}.  For each tracer plugged into the SDK the current tracing span is marked as completed.
+     *
+     * @param context Additional metadata that is passed through the call stack.
+     * @param signal  The signal indicates the status and contains the metadata we need to end the tracing span.
+     */
+    public <T extends CosmosResponse<? extends Resource>> void endSpan(Context context, Signal<T> signal, int statusCode) {
+        Objects.requireNonNull(context, "'context' cannot be null.");
+        Objects.requireNonNull(signal, "'signal' cannot be null.");
+
+        switch (signal.getType()) {
+            case ON_COMPLETE:
+                end(statusCode, null, context);
+                break;
+            case ON_ERROR:
+                Throwable throwable = null;
+                if (signal.hasError()) {
+                    // The last status available is on error, this contains the thrown error.
+                    throwable = signal.getThrowable();
+
+                    if (throwable instanceof CosmosException) {
+                        CosmosException exception = (CosmosException) throwable;
+                        // confirm ?
+                        statusCode = exception.getStatusCode();
+                    }
+                }
+                end(statusCode, throwable, context);
+                break;
+            default:
+                // ON_SUBSCRIBE and ON_NEXT don't have the information to end the span so just return.
+                break;
+        }
+    }
+
+    public <T extends CosmosResponse<?>> Mono<T> traceEnabledCosmosResponsePublisher(Mono<T> resultPublisher,
+                                                                                                             Context context,
+                                                                                                             String spanName,
+                                                                                                             String databaseId,
+                                                                                                             String endpoint) {
+        return traceEnabledPublisher(resultPublisher,  context, spanName,databaseId, endpoint,
+            (T response) -> response.getStatusCode());
+    }
+
+    public <T> Mono<T> traceEnabledNonCosmosResponsePublisher(Mono<T> resultPublisher,
+                                                              Context context,
+                                                              String spanName,
+                                                              String databaseId,
+                                                              String endpoint) {
+        return traceEnabledPublisher(resultPublisher,  context, spanName, databaseId, endpoint, (T response) -> HttpConstants.StatusCodes.OK);
```

Comment:
Why are we always sending a `200 OK` status here?

Reply:
Initially we had non cosmosResponse api returning Mono<Integer>, now we removed those api , so removing this method altogether